Description: Upstream changes introduced in version 4.20.0-0ubuntu2
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 unity (4.20.0-0ubuntu2) oneiric; urgency=low
 .
   * Merge trunk:
     - Desktop drawn with offset (LP: #862743)
 .
 The person named in the Author field signed this changelog entry.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/862743

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- unity-4.20.0.orig/plugins/unityshell/src/HomeView.cpp
+++ unity-4.20.0/plugins/unityshell/src/HomeView.cpp
@@ -36,6 +36,26 @@ namespace
 nux::logging::Logger logger("unity.dash.homeview");
 }
 
+
+ // this is so we can access some protected members in scrollview
+class LensScrollView: public nux::ScrollView
+{
+  protected:
+    nux::VScrollBar* vscrollbar_;
+
+public:
+  LensScrollView()
+    : nux::ScrollView()
+  {
+    vscrollbar_ = NULL;
+  }
+
+  void SetVScrollBar(nux::VScrollBar* newVScrollBar)
+  {
+    nux::ScrollView::SetVScrollBar(newVScrollBar);
+  }
+};
+
 NUX_IMPLEMENT_OBJECT_TYPE(HomeView);
 
 HomeView::HomeView()
@@ -69,12 +89,13 @@ void HomeView::SetupViews()
 {
   layout_ = new nux::HLayout(NUX_TRACKER_LOCATION);
 
-  scroll_view_ = new nux::ScrollView();
+  scroll_view_ = new LensScrollView();
   scroll_view_->EnableVerticalScrollBar(true);
   scroll_view_->EnableHorizontalScrollBar(false);
+  static_cast<LensScrollView *>(scroll_view_)->SetVScrollBar(new PlacesVScrollBar());
   scroll_view_->SetVisible(false);
   layout_->AddView(scroll_view_);
-  
+
   scroll_layout_ = new nux::VLayout();
   scroll_view_->SetLayout(scroll_layout_);
 
@@ -101,7 +122,7 @@ void HomeView::AddLens(Lens::Ptr lens)
   group->expanded.connect(sigc::mem_fun(this, &HomeView::OnGroupExpanded));
   categories_.push_back(group);
   counts_[group] = 0;
-  
+
   ResultViewGrid* grid = new ResultViewGrid(NUX_TRACKER_LOCATION);
   grid->expanded = false;
   grid->SetModelRenderer(new ResultRendererTile(NUX_TRACKER_LOCATION));
--- unity-4.20.0.orig/plugins/unityshell/src/unityshell.cpp
+++ unity-4.20.0/plugins/unityshell/src/unityshell.cpp
@@ -1832,8 +1832,10 @@ bool UnityWindow::place(CompPoint& pos)
 {
   bool result = window->place(pos);
 
-  pos = tryNotIntersectUI(pos);
+  if (window->type() & NO_FOCUS_MASK)
+    return result;
 
+  pos = tryNotIntersectUI(pos);
   return result;
 }
 
--- unity-4.20.0.orig/services/CMakeLists.txt
+++ unity-4.20.0/services/CMakeLists.txt
@@ -39,7 +39,7 @@ set(CFLAGS
     "-DGETTEXT_PACKAGE=\"unity\""
     "-DINDICATORDIR=\"${_indicatordir}\""
     "-DINDICATORICONDIR=\"${_iconsdir}\""
-    "-Werror -Wall"
+    "-Wall"
     )
 add_definitions(${CFLAGS})
 
